internal/pkg/wirecli/wirecli.go

```diff
@@ -18,21 +18,45 @@ import (
 
 	"github.com/google/wire"
 	"sigs.k8s.io/cli-experimental/internal/pkg/apply"
+	"sigs.k8s.io/cli-experimental/internal/pkg/delete"
+	"sigs.k8s.io/cli-experimental/internal/pkg/prune"
 	"sigs.k8s.io/cli-experimental/internal/pkg/status"
+	"sigs.k8s.io/cli-experimental/internal/pkg/wirecli/wireconfig"
 	"sigs.k8s.io/cli-experimental/internal/pkg/wirecli/wiregit"
 	"sigs.k8s.io/cli-experimental/internal/pkg/wirecli/wirek8s"
 )
 
 // ProviderSet defines dependencies for initializing objects
-var ProviderSet = wire.NewSet(wirek8s.ProviderSet, wiregit.OptionalProviderSet,
-	wire.Struct(new(status.Status), "*"), wire.Struct(new(apply.Apply), "*"), NewStatusCommandResult, NewApplyCommandResult)
+var ProviderSet = wire.NewSet(
+	wirek8s.ProviderSet,
+	wiregit.OptionalProviderSet,
+	wire.Struct(new(status.Status), "*"),
+	wire.Struct(new(apply.Apply), "*"),
+	wire.Struct(new(prune.Prune), "*"),
+	wire.Struct(new(delete.Delete), "*"),
+	NewStatusCommandResult,
```

Comment:
Consider per command wire.
Meaning status/wire.go which includes this providerSet
That way the wirecli/wirecli.go and wirecli/wire.go need not change for every subcommand.

similarly apply/wire.go, prune/wire.go

Reply:
Added a TODO to address it in a following PR.

Reply:
done in this PR. added it as the 3rd commit.

Reply:
Thanks.